### PR TITLE
Upload sidevm code via pRPC

### PR DESCRIPTION
**Pull request by kvinwang**
This PR allows deploying sidevm instances in ink without the corresponding codes pre-uploaded.
If the code is not found while deploying, the instance would be set to state `WaitingForCode`. Users can upload sidevm code which matches the code hash later via the added RPC.

### CODE REVIEW

**Comment by h4x3rotab** [comment]
Conceptually good. Now there are multiple ways to deploy SideVM. Should we unify them or reduce the possibilities?

- Deploy by tx, code stored on chain
- Deploy by tx, code uploaded later by rpc
- Deploy by query, code stored on chain
- Deploy by query, code uploaded later by rpc

**Comment by kvinwang** [comment]
> Should we unify them or reduce the possibilities?

I think we show the most two common ways in the examples, no need to add limits.
- The most common way might be:
   `Deploy by query, code uploaded later by rpc` for apps like discord bot.
- The second most common way might be:
   `Deploy by tx, code stored on-chain` for apps like cluster logger.

File: e2e/src/fullstack.js [hunk 1]
```diff
@@ -676,6 +677,21 @@ describe('A full stack', function () {
             }, 4 * 6000), 'Failed to apply deposit to contract weight');
         });
 
+        it('can start sidevm without code uploaded', async function () {
```
Comment (h4x3rotab) [inline e2e/src/fullstack.js:680]:
```suggestion
        it('can set the sidevm as pending state without code uploaded', async function () {
```

**Review by h4x3rotab — APPROVED** [review]
LGTM

**Pull request merged**